nearcore/src/runtime/mod.rs

```diff
@@ -823,14 +823,24 @@ impl RuntimeAdapter for NightshadeRuntime {
 
         // Total amount of gas burnt for converting transactions towards receipts.
         let mut total_gas_burnt = 0;
+        let mut total_size = 0u64;
         // TODO: Update gas limit for transactions
         let transactions_gas_limit = gas_limit / 2;
         let mut transactions = vec![];
         let mut num_checked_transactions = 0;
 
         let runtime_config = self.runtime_config_store.get_config(current_protocol_version);
 
-        while total_gas_burnt < transactions_gas_limit {
+        // In general, we limit the number of transactions via send_fees.
+        // However, as a second line of defense, we want to limit the byte size
+        // of transaction as well. Rather than introducing a separate config for
+        // the limit, we compute it heuristically from the gas limit and the
+        // cost of roundtripping a byte of data through disk.
+        let size_limit = transactions_gas_limit
+            / (runtime_config.wasm_config.ext_costs.storage_write_value_byte
+                + runtime_config.wasm_config.ext_costs.storage_read_value_byte);
+
+        while total_gas_burnt < transactions_gas_limit && total_size < size_limit {
```

Comment:
I wonder if we shouldn't try to make the size limit strict. This way, we check the size limit after we added the transaction. I guess the 4MB per TX limit prevents us from going too far above, so I think it's completely fine if you keep it as it is.

Reply:
Yeah, I considered that. In the end, I decided to follow the existing logic for gas limit.